Ch implement feedback 166087843

Ph3bian opened the pull request:
#### What does this PR do?
Updated loan application to verify if a loan is approved before repayment

Check if a user is verified before loan application complete

Check if a user has an active loan before applying for another loan

Get repayment history for loan complete

Update loan to repaid on final repayment done

Update activeLoan to false on loan repayment completion

#### What are the relevant pivotal tracker stories?
[Finishes #166087843]

Commit:
feature(database): User request loan

[Finishes #166054854]

Commit:
Chore:Updated testing, included status code

Commit:
testing bug on heroku

Commit:
updates

Ph3bian committed:
Merge branch 'develop' into ch-update-testing-166082660

Commit:
updates

Commit:
took out console.logs

Commit:
updates

Commit:
updates

Commit:
Updated error messages and tests

Commit:
chore: Corrected wrong status code

Implemented fetch all loans(from database)
Implemented feature verify user (database)

Standardized error name in catch block

[Finishes #166104614, #166107746, #166079340]

Commit:
Updated Migration file, tests

Ph3bian committed:
Merge branch 'develop' into ch-implement-update-review-2-166104614

Commit:
included babel plugin for try catch

Commit:
updates

Commit:
declared token

Commit:
chore(api): updated swagger.json

Commit:
updated swagger.json

Ph3bian committed:
chore(database): Implement repayments from database
Check if user is verified before request loan
Fetch repayments from database
Create repayments to database
Fixed wildcard route default handler
Updated error message on 500
Implementation of mails began but not completed

Ph3bian committed:
chore(api, database): Implemented repayments

Updated loan application to verify if loan is approved before repayment

Check if user is verified before loan application complete

Check if user has an active loan before applying for another loan

Get repayment history for loan complete

Update loan to repaid on final repayment done

Pivotal tracker:
[Finishes #166087843]

Ph3bian committed:
chore(api, database): Implemented repayments

Updated loan application to verify if loan is approved before repayment

Check if user is verified before loan application complete

Check if user has an active loan before applying for another loan

Get repayment history for loan complete

Update loan to repaid on final repayment done

Pivotal tracker:
[Finishes #166087843]

Ph3bian committed:
resolved conflict

Ph3bian committed:
resolved conflict

oluseyianani commented on server/database/migration.js:
```diff
@@ -27,7 +27,8 @@ CREATE TABLE IF NOT EXISTS users (
     email VARCHAR(255) NOT NULL UNIQUE,
     password VARCHAR(255) NOT NULL,
-    bvn VARCHAR(255) NOT NULL
+    bvn VARCHAR(255) NOT NULL,
```
I thought you said this was going to be optional?

Ph3bian replied:
Fixed now

oluseyianani commented on server/test/tests.js:
```diff
@@ -431,3 +428,35 @@ describe('PATCH /users/<:user-email>/verify', () => {
+    console.log(newUser)
+    const newLoan = await loanModel.create({ ...loanData, userId: newUser.id });
+    await userModel.updateAdminStatus(user.email);
+    console.log(newLoan)
```
Please don't forget to remove this.

Ph3bian replied:
Removed

oluseyianani commented on server/test/tests.js:
```diff
@@ -431,3 +428,35 @@ describe('PATCH /users/<:user-email>/verify', () => {
+
+    const result = await userModel.create(userData);
+    const newUser = result.rows[0];
+    // const token = jwt.sign({ id: user.id }, config.jwtSecret);
```
Please don't forget to remove this.

Ph3bian replied:
got it, removed

oluseyianani commented on server/controllers/authController.js:
```diff
@@ -11,76 +11,91 @@ export default class AuthController {
-            error: 'Invalid email address or password',
-          });
-        }
+    // try {
```
Please don't forget to remove this.

Ph3bian replied:
got it

oluseyianani commented on server/controllers/authController.js:
```diff
@@ -11,76 +11,91 @@ export default class AuthController {
-          error: 'Incorrect login details',
-        });
+    if (bcrypt.compareSync(req.body.password, user.password)) {
+      const token = jwt.sign({ id: user.id }, config.jwtSecret);
```
If you find the time, you can extract this method to avoid repetition on signup and login. It'll be easier to test as a unit test.

Ph3bian replied:
ok, done

oluseyianani reviewed (commented):
Good job on your implementation. You can fix the comment and merge.

oluseyianani commented:
Please fix your failing test before merging

Ph3bian committed:
merged conflicts in updated test file

Ph3bian committed:
chore(api, database): Implemented repayments
Updated loan application to verify if loan is approved before repayment

Check if user is verified before loan application complete

Check if user has an active loan before applying for another loan

Get repayment history for loan complete

Update loan to repaid on final repayment done

Pivotal tracker:
[Finishes #166087843]
